features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/DefaultClassificationService.java

```diff
@@ -79,11 +87,14 @@ public DefaultClassificationService(ClassificationRuleDao classificationRuleDao,
                                         SessionUtils sessionUtils) {
         this.classificationRuleDao = Objects.requireNonNull(classificationRuleDao);
         this.classificationGroupDao = Objects.requireNonNull(classificationGroupDao);
-        this.classificationEngine = Objects.requireNonNull(classificationEngine);
+        this.classificationEngine = new AsyncReloadingClassificationEngine(Objects.requireNonNull(classificationEngine));
```

Comment:
I think the wrapping should be done in the application context wiring.

Reply:
I think it is a matter of taste. The `AsyncReloadingClassificationEngine` was only introduced to support consistent and non-sluggish front-end behavior. It allows to repeatedly trigger reloads but blocks on classification if a reload is on-going.

The other classification engine instances that are configured in the application context (`DefaultClassificationEngine` and `TimingClassificationEngine`) are of general use. In particular, these two classification are non-blocking even in case of on-going reloads.

The `AsyncReloadingClassificationEngine` is only used in the `DefaultClassificationService` that underlies the `ClassificationRestService`.

Reply:
It's currently only used that way. Please do note remove the flexibility we head before.

Reply:
AsyncReloadingClassification is now wired in app context.